src/controller/python/OpCredsBinding.cpp

```diff
@@ -687,4 +748,29 @@ PyChipError pychip_GetCompletionError()
     return ToPyChipError(sTestCommissioner.GetCompletionError());
 }
 
+extern "C" {
+// Define a global callback to set the RCAC data callback
+using PythonRCACCallback                      = void (*)(const uint8_t *, size_t);
```

Comment:
General question - why do this as a callback and a future rather than just a function to get a blob of data that's stored in the test commissioner? The RCAC isn't really valid until the commissioning completes anyway, so is there a point of separating the callbacks?

Reply:
So here's the concern with this - the commissioning delegate is already quite complex and you're introducing a fair amount of new complexity here to just surface the RCAC byte values at the point when they're captured. But there's no reason for any of the tests to attempt to access the RCAC until after the commissioning completes anyway since it's not valid until you know the device has accepted it. 

Does the commissioning process hang if you do nothing but copy out the RCAC to the vector? It shouldn't. I'm concerned what's actually happening here is a premature completion signal based on this new future.

basically, this SHOULD work with just a copy to the vector and none of this callback stuff. If you can get it working at that point, all this callback stuff can be removed and you should be able to retrieve the data after the commissioning completes.

Reply:
Thank you for bringing this up, I had not considered what you have mentioned above previously. I completely agree with you, it is always best to keep the code as simple as possible while completing its purpose as long as it's not causing any additional problems. In this point the callbacks that were previously established were adding complexity and possible issues that were going unnoticed during my testing unknowingly at the time.

When I first attempted implementing the code the hang appeared to have been due to attempting to implement the pulling point of the RCAC/RPK from the OnCommissioningComplete function in the OnNetworkCommissioning function of ChipDeviceCtrl python module. However, since moving that outside the embedded OnNetworkCommissioning function and making sure that the new functionality was implemented correctly, while also providing the expected value back, then we noticed no further hangs or issues when running the new code.

Updated the new code to remove the callbacks and replaced with a similar function that you have mentioned above, attempting to keep the code simple without adding any unnecessary complexity to it that was being caused by the callbacks, this should hopefully remove any underlying issues that could have gone previously unnoticed.